### DISCUSSION TITLE
API OOM!!! but the workflow in the ComfyUI is working.

### DISCUSSION THREAD

**Discussion by monster-echo** [comfyanonymous/ComfyUI#5924] [Q&A]
I have a workflow running in the ComfyUI. 
But When I export it  as API json, and invoke it in a python console, it will OOM at the second invoke. 

It seems like the comfyUI reuse the VRAM when running a same workflow or other mechanism, the VRAM is not cleaned waiting the next invoked.

But with the API, the reserved VRAM is always not reused. 

Here is the python console code 

``` python
import json
import requests
from ollama import random_prompt
import time


with open("workflows/1.json", "r", encoding="utf-8") as f:
    data = json.loads(f.read())


headers = {
    'Content-Type': 'application/json',
    'Cache-Control': 'no-cache',
    'Comfy-User': ''
}

session = requests.Session()
for i in range(5000):
    print(f"iteration {i}")
    data["prompt"]["3"]["inputs"]["text"] = random_prompt("crazy weird art, All kinds of scenarios, like a dream or a nightmare.")
    resp = session.post("http://localhost:8188/api/prompt", json=data, headers=headers)
    print(resp.json())
    time.sleep(70)
    print("done")
```

<img width="866" alt="image" src="https://github.com/user-attachments/assets/48b911c4-beac-4cec-8403-039ddabebe49">

**Comment by monster-echo** [comment]
I reproduced this code in a react project. At first it's OOM too.

But when I add proxy to the http://127.0.0.1:8188/ with the Vite server configuration,  It's worked. 

Can't figure out this mechanism , could you please help me out, thanks.

**Comment by ltdrdata** [comment]
Try with `--disable-smart-memory`
